thrust/examples/include/host_device.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *  Copyright 2008-2009 NVIDIA Corporation
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#pragma once
+
+#if THRUST_DEVICE_COMPILER != THRUST_DEVICE_COMPILER_NVCC
+
+#  ifndef __host__
```

Comment:
We should not do that, but rather use `_CCCL_HOST` or `_CCCL_DEVICE` or `_CCCL_HOST_DEVICE` or at least define a proper macro `TEST_HOST_DEVICE` that evaluates to the internal one

Reply:
I've updated tests and sources to use our internal macros, but this header is only included from example sources where using `__host__ __device__` instead of an internal macro is preferable, and these also need to build for Thrust's non-CUDA backends.